core/modules/admin_bar/admin_bar.inc

```diff
@@ -78,6 +78,11 @@ function admin_bar_output($complete = FALSE) {
       $content['extra']['extra'] += admin_bar_links_users();
     }
 
+    // Check for status report errors.
+    if (user_access('administer site configuration')) {
+      $content['extra']['extra'] += admin_bar_links_alert();
```

Comment:
Looks like we're still adding the alert on the right side in "extra" instead of being in the menu like you described in https://github.com/backdrop/backdrop-issues/issues/639#issuecomment-171865841.